cli/src/main/java/org/jboss/as/cli/handlers/ShutdownHandler.java

```diff
@@ -172,28 +171,39 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
             disconnect = false;
         }
 
+        // Check if we are using a client launched from the server/host installation we want shutdown to perform an update.
+        // In such a case, we will exit from the current JBoss CLI process to avoid interfering with the server/host update.
+        // This will also force the user to relaunch the CLI session using the most recent updates once the server/host has
+        // been updated.
+        // The shutdown mgmt Operation will return the content of the same client marker file created by this CLI instance if
+        // both are using the same server installation. Otherwise, no file marker value will be returned by the ShutDown operations.
+        // This check is only relevant when we are performing a installation.
+        ModelNode clientMarker = executeOperation(client, cliClient, op, true);
         if (Util.TRUE.equalsIgnoreCase(performInstallation.getValue(ctx.getParsedCommandLine()))) {
-            // Check if I am using a client launched from the server/host installation we want to shutdown to perform an update.
-            // In such a case, we will exit from the current JBoss CLI process to avoid interfering with the server/host update.
-            // This will also force the user to relaunch the CLI session using the most recent updates once the server/host has
-            // been updated.
-            boolean localClientLaunch = isLocalClientLaunch(ctx);
-            executeOperation(client, cliClient, op, !localClientLaunch);
-            if (localClientLaunch) {
+            boolean isLocalClient = true;
+            if (clientMarker != null) {
+                final String clientMarkerData = WildFlySecurityManager.getPropertyPrivileged(Util.CLI_MARKER_VALUE, null);
+                if (clientMarkerData != null) {
+                    if (!clientMarker.asString().equals(clientMarkerData)) {
+                        isLocalClient = false;
+                    }
+                }
+            } else {
+                isLocalClient = false;
+            }
+
+            if(isLocalClient) {
                 ctx.printLine("The JBoss CLI session will be closed automatically to allow the server be updated. Once the server has been restarted, you can relaunch the JBoss CLI session.", false);
                 try {
                     TimeUnit.SECONDS.sleep(3);
                 } catch (InterruptedException e) {
                     // Ignored
                 }
-
                 // We are using a CLI which was launched from the server installation we have requested to be updated.
                 // In order to prevent keeping using a jboss-modules.jar that could have been updated, we finish the CLI process
                 // Once the server has been restarted the user will launch again the CLI that will use the most recent updates
                 ctx.terminateSession();
```

Comment:
Shouldn't you return after this call?

Reply:
I've never found an issue, but yes, looks correct returning here.